add info about warehouse snake_casing (#4242)
segmentio/segment-docs: src/connections/storage/warehouses/schema.md

---

A **schema** describes the way that the data in a warehouse is organized. Segment stores data in relational schemas, which organize data into the following template:

> `<source>.<collection>.<property>`, for example `segment_engineering.tracks.user_id`, where source refers to the source or project name (segment_engineering), collection refers to the event (tracks), and the property refers to the data being collected (user_id). All schemas convert collection and property names from `CamelCase` to `snake_case`. For transparency, Segment uses this package to `snake_case` fields in the warehouse pipeline: [go-snakecase]([url](https://github.com/segmentio/go-snakecase)).

Reviewer: ```suggestion
`<source>.<collection>.<property>`, for example `segment_engineering.tracks.user_id`, where source refers to the source or project name (segment_engineering), collection refers to the event (tracks), and the property refers to the data being collected (user_id). All schemas convert collection and property names from `CamelCase` to `snake_case` using the [go-snakecase](https://github.com/segmentio/go-snakecase) package.
```
